### PR TITLE
Update dependencies (preparing 0.50 release)

### PR DESCRIPTION
**Pull request by paulfantom**
## Description

_Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.
If it fixes a bug or resolves a feature request, be sure to link to that issue._

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [ ] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [x] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by paulfantom** [comment]
Turns out images used in e2e tests were built with different golang version (1.15) which caused errors when updating to newer k8s client libs. This PR is fixing that issue and pinning to use golang 1.16 in `go.mod`.

File: .github/workflows/e2e.yaml [hunk 1]
```diff
@@ -33,6 +34,13 @@ jobs:
             thanosruler: ""
     steps:
     - uses: actions/checkout@v2
+    - uses: actions/setup-go@v2
+      with:
+        go-version: ${{ env.golang-version }}
+    - name: Build images
```
Comment (paulfantom) [inline .github/workflows/e2e.yaml:40]: Building images first will improve time-to-failure in case the code is incorrect.

**Review by simonpasquier — APPROVED** [review]
lgtm

**Comment by paulfantom** [comment]
E2E tests are failing due to https://github.com/prometheus-operator/prometheus-operator/issues/4070. Workaround is in #4214

**Pull request merged**